src/solvers/flattening/boolbv_let.cpp

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************\
+
+Module:
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+#include "boolbv.h"
+
+#include <util/std_expr.h>
+#include <util/std_types.h>
+
+bvt boolbvt::convert_let(const let_exprt &expr)
+{
+  const bvt value_bv=convert_bv(expr.value());
+
+  // We expect the bindings to be unique, and
+  // thus, these can go straight into the symbol to literal map.
+  // This property also allows us to cache any subexpressions.
+  const irep_idt &id=expr.symbol().get_identifier();
+
+  map.set_literals(id, expr.symbol().type(), value_bv);
+  bvt result_bv=convert_bv(expr.where());
+  map.erase_literals(id, expr.symbol().type());
```

Comment:
Why are the first created/set and then erased?!

Reply:
Added a comment. The symbol is meant to be visible during the recursive call only.